library/bigip_software.py

```diff
@@ -212,673 +290,1060 @@
+    def is_activated(self):
+        result = self.software_on_volume()
+        if result:
+            self.have, volume = result
+            if hasattr(volume, 'active') and volume.active is True:
+                return True
 
-            while not done:
-                text = base64.b64encode(fh.read(CHUNK_SIZE))
+    def installed(self):
+        if self.is_installed():
+            return False
+        else:
+            self.install_volume()
+            return True
 
-                if first:
-                    if len(text) < CHUNK_SIZE:
-                        chain_type = 'FILE_FIRST_AND_LAST'
-                    else:
-                        chain_type = 'FILE_FIRST'
-                    first = False
-                else:
-                    if len(text) < CHUNK_SIZE:
-                        chain_type = 'FILE_LAST'
-                        done = True
-                    else:
-                        chain_type = 'FILE_MIDDLE'
-
-                self.api.System.ConfigSync.upload_file(
-                    file_name=remote_path,
-                    file_context=dict(
-                        file_data=text,
-                        chain_type=chain_type
-                    )
-                )
+    def is_installed(self):
+        result = self.software_on_volume()
+        if result:
+            self.have, volume = result
+            if not hasattr(volume, 'active') or volume.active is not True:
+                return True
 
-    def is_hotfix_available(self, hotfix):
-        hotfix = os.path.basename(hotfix)
-        images = self.api.System.SoftwareManagement.get_software_hotfix_list()
-        for image in images:
-            if image['filename'] == hotfix:
+    def present(self):
+        if self.exists():
+            return False
+        else:
+            self._set_changed_options()
+            if self.client.check_mode:
                 return True
-        return False
+            self.upload()
+            return True
 
-    def is_software_available(self, software):
-        software = os.path.basename(software)
-        images = self.api.System.SoftwareManagement.get_software_image_list()
-        for image in images:
-            if image['filename'] == software:
+    def absent(self):
+        if not self.exists():
+            return False
+        else:
+            if self.client.check_mode:
                 return True
-        return False
+            self.remove()
+            return True
 
-    def delete(self, software):
-        software = os.path.basename(software)
-        self.api.System.SoftwareManagement.delete_software_image(
-            image_filenames=[software]
-        )
+    def install_volume(self):
+        self._set_changed_options()
+        if self.client.check_mode:
+            return True
+        if not self.exists():
+            self.upload()
+        if self.want.delete_volume:
+            self.delete_volume_on_device()
+        if self.want.hotfix:
```

Comment:
test the negation  you do on line 825 and return early. then backspace all of the code that follows here